### ISSUE TITLE
pytest --collect-only needs a one char shortcut command

### ISSUE THREAD

**Issue by ssbarnea** [pytest-dev/pytest#5845]
I find myself needing to run `--collect-only` very often and that cli argument is a very long to type one. 

I do think that it would be great to allocate a character for it, not sure which one yet. Please use up/down thumbs to vote if you would find it useful or not and eventually proposing which char should be used. 

Clearly this is a change very easy to implement but first I want to see if others would find it useful or not.

**Comment by The-Compiler** [comment]
Agreed, it's probably the option I use most which doesn't have a shortcut.

Both `-c` and `-o` are taken. I guess `-n` (as in "no action", compare `-n`/`--dry-run` for e.g. `git clean`) could work? 

Maybe `--co` (for either "*co*llect" or "*c*ollect *o*nly), similar to other two-character shortcuts we already have (`--sw`, `--lf`, `--ff`, `--nf`)?

**Comment by nicoddemus** [comment]
I like `--co`, and it doesn't seem to be used by any plugins as far as I can search:

https://github.com/search?utf8=%E2%9C%93&q=--co+language%3APython+pytest+language%3APython+language%3APython&type=Code&ref=advsearch&l=Python&l=Python

**Comment by blueyed** [comment]
> I find myself needing to run `--collect-only` very often and that cli argument is a very long to type one.

Just out of curiosity: Why?  (i.e. what's your use case?)

+0 for `--co`.

But in general you can easily also have an alias "alias pco='pytest --collect-only'" - (or "alias pco='p --collect-only" if you have a shortcut for pytest already.. :))

**Comment by twblamer** [comment]
I routinely use `--collect-only` when I switch to a different development branch or start working on a different area of our code base. I think `--co` is fine.

**Comment by NNRepos** [comment]
hey guys I'm trying to make a pull request but I'm not sure how to make and run tests from the src directory:
py -3 -m _pytest.main
...
ModuleNotFoundError: No module named '_pytest._version'

help would be appreciated

**Comment by ssbarnea** [comment]
I am currently using the alias proposed. I am a bit worried that a double dash shortcut is confusing. If we have no free single letters we are better if with alias.

**Comment by NNRepos** [comment]
> I am currently using the alias proposed. I am a bit worried that a double dash shortcut is confusing. If we have no free single letters we are better if with alias.

sure, I'm just making a pull request in case this gets accepted. I'll change accordingly to whatever is proposed eventually.

**Comment by bluetech** [comment]
@ssbarnea pytest already has an established conventions for such shortcuts, in particular

```
  --lf, --last-failed   rerun only the tests that failed at the last run (or all if none failed)
  --ff, --failed-first  run all tests but run the last failures first. This may re-order tests and thus lead to repeated fixture setup/teardown
  --nf, --new-first     run tests from new files first, then the rest of the tests sorted by file mtime
  --lfnf={all,none}, --last-failed-no-failures={all,none}
                        which tests to run with no previously (known) failures.
  --sw, --stepwise      exit on test failure and continue from last failing test next time
```

I think `--co` is a good addition.
